File: src/main/java/rk/cinema/model/Hall.java
package rk.cinema.model;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

@MappedEntity
public record Hall(
        @Id
        @GeneratedValue
        Long id,

        @NonNull
        String name
) {
}

File: src/main/kotlin/rk/powermilk/cinema/model/Hall.kt
package rk.powermilk.cinema.model

import io.micronaut.core.annotation.NonNull
import io.micronaut.serde.annotation.Serdeable
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id

@Serdeable
@Entity
data class Hall(
    @field:Id
    @field:GeneratedValue
    val id: Long = 0,

    val name: @NonNull String,
)



